### ISSUE TITLE
Make buffer flushes faster

### ISSUE THREAD

**Issue by Fingercomp** [cc-ru/wonderful#22]
The `Framebuffer:flush` is too slow, and we basically need to make it faster.

**Comment by Fingercomp** [comment]
The framebuffer divides the whole area into square blocks (or rectangular if they happen to be on the right/bottom edge), and assigns something called "dirtiness" to each of them. It's incremented if a cell in the block gets dirty (changed since the last flush). The flush method iterates over the blocks, skipping those whose dirtiness is 0 (meaning they didn't change since the last flush at all), and compiles instructions to render changes in each of them.

Well, in the [`no-blocks` branch](https://github.com/cc-ru/wonderful/tree/no-blocks), I tried to remove the concept of blocks and operate over the whole area. I had to get rid of fills as well, unfortunately. (If the branch is merged, we will probably need some code that finds the rectangles to `gpu.fill`. Or we could leave it as is. I don't know.)

The effect of this change was a little suprising to me. The code got cleaner.

...

I mean, the master version flushes in 1.6 seconds on my emulator; the no-blocks version manages to bring it down to 0.9 s or so. Therefore, I think I'll merge this branch as soon as we compare the perfomance on an in-game computer. In the meanwhile I need to find a way to convince myself that we can do without fills for some time.
